### ISSUE TITLE
naming-convention: Can't specify rule for specific property name

### ISSUE THREAD

**Issue by jens-duttke** [typescript-eslint/typescript-eslint#1538]
The documentation of the [`naming-convention`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md) rule, is very unclear how to use `filter` and `custom`.

**Repro**

I have this JSX code:
```tsx
<p dangerouslySetInnerHTML={{ __html: "..." }} />
```

And the `naming-convention` rules always complains, that the code either does not follow the format, or that my rule configuration is invalid, if I only specify `custom`, without `format`.

> Property name __html must match one of the following formats: snake_case [Error/@typescript-eslint/naming-convention]

My definition:
```js
{
  "rules": {
    '@typescript-eslint/naming-convention': ['error',
      // Property
      { selector: 'property', format: ['camelCase', 'PascalCase'] },
      {
        selector: 'property',
        filter: '^__html$',
        types: ['string'],
        custom: { regex: '^__html$', match: true },
        format: ['snake_case'],
        leadingUnderscore: 'require'
      }
    }
  }
```

Can someone tell me how to configure this rule correctly, and maybe update the rule documentation to make the usage more clear?

**Additional Info**

My requirements are:
- Properties are either camelCase or PascalCase
- The property name `__name` is also allowed, if the value is a string

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.18.0` |
| `@typescript-eslint/parser`        | `2.18.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.6.0` |
| `npm`                              | `6.13.6` |

**Comment by bradzacher** [comment]
> is very unclear how to use filter and custom.

From the docs:
> filter accepts a regular expression (anything accepted into new RegExp(filter)). It allows you to limit the scope of this configuration to names that match this regex.

Could you please point out what's unclear so I can update the docs if needs be.

---

The rule complains that the name is invalid because the name doesn't match any of the defined formats.

It's not in snake case, because it contains upper case characters.
It's not in camelCase, because it contains an underscore.
--- `leadingUnderscore` only checks for exactly one underscore.
Sorry, the docs are slightly ambiguous due to the plural used due to the wording; "leading or trailing underscores".

Because it doesn't match a format, and never can match one of the formats, you can just pass in `null`

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#format

> Instead of an array, you may also pass null. This signifies "this selector shall not have its format checked".

If you configure it like this, it will do what you want.

```ts
     {
        selector: 'property',
        filter: '^__html$',
        format: null,
     }
```

This config says "when a property named `__html` is found, don't check the format for it".
You need not check the format, because the fact that it has matched the selector suggests it's correctly named.

**Comment by jens-duttke** [comment]
> ```js
>      {
>         selector: 'property',
>         filter: '^__html$',
>         format: null,
>      }
> ```

@bradzacher Thanks! That example is exactly what I was lookling for.

> Could you please point out what's unclear so I can update the docs if needs be.

Your example above, to specify specific names would be a nice addition to the documentation.
